include/engine/guidance/step_maneuver.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef ENGINE_GUIDANCE_STEP_MANEUVER_HPP
+#define ENGINE_GUIDANCE_STEP_MANEUVER_HPP
+
+#include "util/coordinate.hpp"
+#include "extractor/turn_instructions.hpp"
+
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+
+struct StepManeuver
+{
+    util::FixedPointCoordinate location;
+    double heading_before;
+    double heading_after;
+    extractor::TurnInstruction instruction;
+};
```

Comment:
Would love to have the opportunity to add something like a hint to every maneuver. Essentially something like **Turn right at name**, or **turn right onto ramp**